crates/lovely-core/src/patch/regex.rs

```diff
@@ -94,21 +143,84 @@ impl RegexPatch {
                 &mut line_prepend
             );
 
+            // Cleanup and convert the specified root capture to a span.
+            let target_group = {
+                let group_name = self
+                    .root_capture
+                    .as_deref()
+                    .unwrap_or("0")
+                    .replace('$', "");
+
+                if let Ok(idx) = group_name.parse::<usize>() {
+                    groups.get_group(idx)
+                        .unwrap_or_else(|| 
+                            panic!("The capture group at index {idx} could not be found in '{base_str}' with the Regex '{}'", self.pattern))
+                } else {
+                    groups.get_group_by_name(&group_name)
+                        .unwrap_or_else(|| 
+                            panic!("The capture group with name '{group_name}' could not be found in '{base_str}' with the Regex '{}'", self.pattern))
+                }
+            };
+
+            let target_start = (target_group.start as isize + delta) as usize;
+            let target_end = (target_group.end as isize + delta) as usize;
+
+            let mut new_payload = String::new();
+
+            // If left border of insertion is a wordchar -> non-wordchar 
+            // boundary and our patch starts with a wordchar, prepend space so 
+            // it doesn't unintentionally concatenate with characters to its 
+            // left to create a larger identifier.
+            if self.line_prepend.is_empty() && self.payload.starts_with(|x: char| x.is_ascii_alphanumeric() || x == '_' || x == '$' || x == '{' || x == '}') ||
+                self.line_prepend.starts_with(|x: char| x.is_ascii_alphanumeric() || x == '_' || x == '$' || x == '{' || x == '}')
```

Comment:
Simplify this if statement via assigning these values to bool vars - `let is_empty = self.line_prepend.is_empty()` + `if is_empty && ...`, for example. This change will make it easier for new contributors to understand what's going on.

Reply:
It would also be good to strip out these sentinel char values into a vec to reduce code duplication.

Reply:
I've factored out this character check into an internal function.